schema/pod.go

```diff
@@ -116,6 +116,11 @@ func (al AppList) assertValid() error {
 			return fmt.Errorf(`duplicate apps of name %q`, a.Name)
 		}
 		seen[a.Name] = true
+		if a.App != nil {
```

Comment:
I'm not sure about every single detail, but just by reading the existing code this is my understanding of the flow:
 1. TestAppList directly builds its own [RuntimeApp instances](https://github.com/appc/spec/blob/9bdac40da49bfee6737cd4e267c41c201a60f321/schema/pod_test.go#L44) without any `App` member.
 2. [Validation](https://github.com/lucab/spec/blob/b183ae94f9bdd4c8e0e590ecd2f0ff298fd79967/schema/pod.go#L106) is invoked before marshaling
 3. `App` is `nil` at this point, and [marshaling is called on it](https://github.com/lucab/spec/blob/b183ae94f9bdd4c8e0e590ecd2f0ff298fd79967/schema/pod.go#L120)
 4. panic due to nil pointer dereference

> (why is .App being dereferenced? the go marshaller should skip nil pointers)

This is a validation routine, used for sanity checks both for marshaling and unmarshaling. In the specific marshaling codepath, validation happens *before* marshaling (marshaller will later skip nil pointers, but validator panics on them first).